src/containers/ImportResources/ImportResources.js

```diff
@@ -100,9 +119,18 @@ export class ImportResources extends Component {
     const gitresourcename = 'git-source';
     const gitcommit = 'master';
 
-    if (repourl === '' || !namespace) {
+    const valid = validateURL(repourl);
+    if (valid === false || !namespace) {
+      if (valid === false && repourl === '') {
+        this.setState({
+          invalidInput: repourl === ''
+        });
+      } else if (valid === false) {
+        this.setState({
+          invalidInput: repourl
```

Comment:
How does this work?

Reply:
It was a situation when it wouldnt return both errors about namespace and repourl without them being in the same loop
It checking that if the repourl has "valid" text in and if not and its not empty (as ifs its empty it doesnt work with setting invalidInput to repourl and if it has text in doesnt show the error on the dashboard with setting it to the state it sets it in the above if statement) then it sets it to invalidInput and errors on the page

Reply:
Thanks, yeah I'd have no idea it does that from the code though, can you add comments to that effect please and see if you can refactor to make things any easier to maintain?

repourl presumably evaluates to true or false then? I don't mind if we add a few extra lines if it makes the code much clearer

Reply:
Yeah I'll add some comments in 
I'll see if i can refactor to make it better to look at / more maintainable 

repourl is the text inputted into the repository url field, thats what get used in validateURL 
valid is the result of the validateURL function which goes either yep valid or nope not valid

Reply:
Great, thanks Carolyn that'd be useful 👍